packages/x-charts-pro/src/internals/plugins/useChartProZoom/useChartProZoom.ts

```diff
@@ -168,54 +163,72 @@ export const useChartProZoom: ChartPlugin<UseChartProZoomSignature> = ({
+          const movement = options.axisDirection === 'x' ? movementX : movementY;
+          const dimension = options.axisDirection === 'x' ? drawingArea.width : drawingArea.height;
+          let newMinPercent = min - (movement / dimension) * span;
+          let newMaxPercent = max - (movement / dimension) * span;
+          if (newMinPercent < MIN_PERCENT) {
+            newMinPercent = MIN_PERCENT;
+            newMaxPercent = newMinPercent + span;
+          }
+          if (newMaxPercent > MAX_PERCENT) {
+            newMaxPercent = MAX_PERCENT;
+            newMinPercent = newMaxPercent - span;
+          }
+          if (
+            newMinPercent < MIN_PERCENT ||
+            newMaxPercent > MAX_PERCENT ||
+            span < options.minSpan ||
+            span > options.maxSpan
+          ) {
+            return zoom;
+          }
+          return {
+            ...zoom,
+            start: newMinPercent,
+            end: newMaxPercent,
+          };
+        });
+        setZoomDataCallback(newZoomData);
+      },
+    );
+
     const handlePan = (event: PointerEvent) => {
       if (element === null || !isDraggingRef.current || panningEventCacheRef.current.length > 1) {
         return;
       }
       if (touchStartRef.current == null) {
         return;
       }
+
+      const touchStart = touchStartRef.current;
       const point = getSVGPoint(element, event);
-      const movementX = point.x - touchStartRef.current.x;
-      const movementY = (point.y - touchStartRef.current.y) * -1;
-      const newZoomData = touchStartRef.current.zoomData.map((zoom) => {
-        const options = optionsLookup[zoom.axisId];
-        if (!options || !options.panning) {
-          return zoom;
-        }
-        const min = zoom.start;
-        const max = zoom.end;
-        const span = max - min;
-        const MIN_PERCENT = options.minStart;
-        const MAX_PERCENT = options.maxEnd;
-        const movement = options.axisDirection === 'x' ? movementX : movementY;
-        const dimension = options.axisDirection === 'x' ? drawingArea.width : drawingArea.height;
-        let newMinPercent = min - (movement / dimension) * span;
-        let newMaxPercent = max - (movement / dimension) * span;
-        if (newMinPercent < MIN_PERCENT) {
-          newMinPercent = MIN_PERCENT;
-          newMaxPercent = newMinPercent + span;
-        }
-        if (newMaxPercent > MAX_PERCENT) {
-          newMaxPercent = MAX_PERCENT;
-          newMinPercent = newMaxPercent - span;
-        }
-        if (
-          newMinPercent < MIN_PERCENT ||
-          newMaxPercent > MAX_PERCENT ||
-          span < options.minSpan ||
-          span > options.maxSpan
-        ) {
-          return zoom;
-        }
-        return {
-          ...zoom,
-          start: newMinPercent,
-          end: newMaxPercent,
-        };
-      });
-      setZoomDataCallback(newZoomData);
+
+      throttledHandlePan(touchStart, point);
```

Comment:
Need to do this instead of wrapping `handlePan` in `rafThrottle` because otherwise the `pointerup` event handler might have been called before the throttled `handlePan`, causing the event handler to be discarded (because it depends on `panningEventCacheRef.current.length`).